hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3217,6 +3218,9 @@ func (r *HostedClusterReconciler) delete(ctx context.Context, hc *hyperv1.Hosted
 
 		if od, ok := p.(platform.OrphanDeleter); ok {
 			if err = od.DeleteOrphanedMachines(ctx, r.Client, hc, controlPlaneNamespace); err != nil {
+				if _, ok := err.(utilerrors.Aggregate); ok {
+					skippedCloudResourcesDeletion.WithLabelValues(hc.Name).Set(float64(1))
```

Comment:
why are you casting the error here?
Also, I think if `DeleteOrphanedMachines` returned an error, then it means we haven't actually skipped resource deletion yet.

you were recording the metric in this line https://github.com/openshift/hypershift/blob/2860ed5d7d4ac3559105a7ff3fdebc5022f0642e/hypershift-operator/controllers/hostedcluster/internal/platform/aws/aws.go#L323 before, which I think is the correct place.